polygon/sync/sync_to_tip_events.go

```diff
@@ -0,0 +1,31 @@
+package sync
+
+import (
+	"github.com/ledgerwatch/erigon/core/types"
+	"github.com/ledgerwatch/erigon/polygon/heimdall"
+	"github.com/ledgerwatch/erigon/polygon/p2p"
+)
+
+const EventTypeMilestone = "milestone"
+const EventTypeNewHeader = "new-header"
+
+type Event struct {
+	Type string
+
+	Milestone *heimdall.Milestone
```

Comment:
btw how about we use generics?

```
type Event[T any] struct {
    Data T
}
```

Reply:
sure. let's address this in the next PR that will implement SyncToTipEvents